### PR TITLE
Hinting review

### PR DESCRIPTION
**Pull request by pferrel**
Still have the ML model persistence to implement for Harness restart but functional tests are successful in a single Harness session.

### CODE REVIEW

File: rest-server/templates/src/main/scala/com/actionml/templates/navhinting/NavHintingAlgorithm.scala [hunk 1]
```diff
@@ -0,0 +1,219 @@
...
+  /** add the event to the end of an active journey subject to length limits */
+  def updateTrail(navId: String, timeStamp: DateTime, trail: Seq[(String, DateTime)]): Seq[(String, DateTime)] = {
+    val newTrail = trail :+ (navId, timeStamp)
+    newTrail.takeRight(params.numQueueEvents.getOrElse(50))
+  }
+
+
+  /** update the model with a Future for every input. This may cause Futures to accumulate */
+  def updateModel(convertedJourney: Journey,  now: DateTime): Unit = {
+    applyDecayFunction(convertedJourney, now).map { weightedVectors =>
```
Comment (pferrel) [inline rest-server/templates/src/main/scala/com/actionml/templates/navhinting/NavHintingAlgorithm.scala:116]: this maps a Future

Comment (pferrel) [reply rest-server/templates/src/main/scala/com/actionml/templates/navhinting/NavHintingAlgorithm.scala:116]: Do we have the possibility to have overlapping Futures here, both modifying the `var model`?

Comment (andreyshikov) [reply rest-server/templates/src/main/scala/com/actionml/templates/navhinting/NavHintingAlgorithm.scala:116]: summarizing slack conversation: we can combine these two futures and return resulting future to the upper level. e.g. like that:
`def updateModel(...): Future[Unit] = {
    applyDecayFunction.map( <merge operation> ).flatMap(<write model file>)`

Comment (pferrel) [reply rest-server/templates/src/main/scala/com/actionml/templates/navhinting/NavHintingAlgorithm.scala:116]: done, it will show up in the next PR

**Pull request merged**